src/utils.jl

```diff
@@ -1,12 +1,34 @@
 eltypes(::Type{T}) where {T} = map_params(eltype, T)
 
+"""
+    StructArrays.map_params(f, T)
+
+Apply `f` to each field type of `Tuple` or `NamedTuple` type `T`, returning a
+new `Tuple` or `NamedTuple` type.
+
+```julia-repl
+julia> StructArrays.map_params(T -> Complex{T}, Tuple{Int32,Float64})
+Tuple{Complex{Int32},Complex{Float64}}
+```
+"""
 map_params(f, ::Type{Tuple{}}) = Tuple{}
 function map_params(f, ::Type{T}) where {T<:Tuple}
     tuple_type_cons(f(tuple_type_head(T)), map_params(f, tuple_type_tail(T)))
 end
 map_params(f, ::Type{NamedTuple{names, types}}) where {names, types} =
     NamedTuple{names, map_params(f, types)}
 
+"""
+    StructArrays._map_params(f, T)
```

Comment:
Now that it is documented, it feels really odd that the API to map a function to a `(Named)Tuple` type returns a type or an object depending on an underscore... Even though it's internal, it may be worth to rename it (don't have great ideas though), but that should probably be done in another PR.